lib/connect.js

```diff
@@ -234,25 +240,86 @@ const getTasks = (model, setupListeners) => {
         setupListeners(mongoClient);
       }
 
+      let maybeTunnelError;
+
+      if (tunnel) {
+        tunnel.on('error', (tunnelError) => {
+          if (tunnelError.origin === 'ssh-client') {
+            maybeTunnelError = tunnelError;
+          }
+        });
+      }
+
       mongoClient.connect((err, _client) => {
+        // When mongo client failed to connect AND tunnel encountered an error
+        // those two are totally related and we want to return a tunnel error as
+        // it will have more relevant context for the end user
+        err = err && maybeTunnelError ? maybeTunnelError : err;
```

Comment:
there are better ways to do this if with Promises

Reply:
There is a chance then that we resolve before either tunnel or client cleaned up after itself, in current implementation, no matter what happens we are sure that every callback resolved before returning

Reply:
Refactored in 7e1ac00